src/main/java/build/buildfarm/instance/server/AbstractServerInstance.java

```diff
@@ -326,7 +329,15 @@ public DigestUtil getDigestUtil() {
             directExecutor()));
   }
 
-  private static boolean shouldEnsureOutputsPresent(RequestMetadata requestMetadata) {
+  private static boolean shouldEnsureOutputsPresent(
```

Comment:
Can you make this a non-static method to avoid having to plumb through a boolean? Or are you doing this to make testing easier?

Reply:
that would probably look cleaner.  I've been preserving `static` so that the code can be easily moved out of the class in a future refactor (and tested as you say).  Currently not unit tested.